src/main/webapp/app/shared/sticky-popover/sticky-popover.directive.ts

```diff
@@ -0,0 +1,104 @@
+import {
+    ApplicationRef,
+    ChangeDetectorRef,
+    Directive,
+    ElementRef,
+    Inject,
+    Injector,
+    Input,
+    NgZone,
+    OnDestroy,
+    OnInit,
+    Renderer2,
+    TemplateRef,
+    ViewContainerRef,
+} from '@angular/core';
+import { DOCUMENT } from '@angular/common';
+import { NgbPopover, NgbPopoverConfig } from '@ng-bootstrap/ng-bootstrap';
+
+@Directive({
+    selector: '[jhiStickyPopover]',
+})
+export class StickyPopoverDirective extends NgbPopover implements OnInit, OnDestroy {
```

Comment:
This new directive looks a bit complex. Can we just use NgbPopover instead or is there a good reason for a custom one? :)

Reply:
NgbPopover can be opened and closed on a hover event. However, when you move your cursor away from the hoverable component/icon to the popover (e.g. to copy some text) the popover closes. This directive is reusable and allows us to provide popovers that contain interactable components without the requirement of a closing-button that we currently use (e.g. for cloning repositories in programming exercises).

Reply:
Can you check how it's done for the [ThemeSwitchComponent](https://github.com/ls1intum/Artemis/blob/8d51aec9e807681cca93602a7739944da1e21971/src/main/webapp/app/core/theme/theme-switch.component.ts)?
If you move the mouse away from the hoverable component there (the navbar sun icon), it does not close if your mouse remains on the popover. You can interact with it, in this case to toggle the OS sync switch, but you can also copy text.


https://github.com/ls1intum/Artemis/assets/23171488/567183df-2b27-4668-8cf7-1403c52ad38d